File: app/src/main/java/vn/loitp/app/activity/function/MenuFunctionActivity.java
package vn.loitp.app.activity.function;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import com.core.base.BaseFontActivity;
import com.core.utilities.LActivityUtil;

import loitp.basemaster.R;
import vn.loitp.app.activity.function.activityandservice.ActivityServiceComunicateActivity;
import vn.loitp.app.activity.function.dragdropsample.DragDropSampleActivity;
import vn.loitp.app.activity.function.fullscreen.FullScreenActivity;
import vn.loitp.app.activity.function.gesto.GestoActivity;
import vn.loitp.app.activity.function.glide.GlideActivity;
import vn.loitp.app.activity.function.hashmap.HashMapActivity;
import vn.loitp.app.activity.function.keyboard.KeyboardActivity;
import vn.loitp.app.activity.function.location.LocationActivity;
import vn.loitp.app.activity.function.notification.MenuNotificationActivity;
import vn.loitp.app.activity.function.recolor.RecolorActivity;
import vn.loitp.app.activity.function.sensor.SensorActivity;
import vn.loitp.app.activity.function.simplefingergestures.SimpleFingerGesturesActivity;
import vn.loitp.app.activity.function.viewdraghelper.ViewDragHelperActivity;
import vn.loitp.app.activity.function.viewdraghelpersimple.ViewDragHelperSimpleActivity;
import vn.loitp.app.activity.function.viewdraghelpersimple.ViewDragHelperSimpleActivity1;

public class MenuFunctionActivity extends BaseFontActivity implements View.OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setShowAdWhenExit(false);

        findViewById(R.id.bt_gesto).setOnClickListener(this);
        findViewById(R.id.bt_simple_finger_gesture).setOnClickListener(this);
        findViewById(R.id.bt_hashmap).setOnClickListener(this);
        findViewById(R.id.bt_drag_drop_sample).setOnClickListener(this);
        findViewById(R.id.bt_toggle_fullscreen).setOnClickListener(this);
        findViewById(R.id.bt_view_drag_helper).setOnClickListener(this);
        findViewById(R.id.bt_recolor).setOnClickListener(this);
        findViewById(R.id.bt_activity_service_comunicate).setOnClickListener(this);
        findViewById(R.id.bt_location).setOnClickListener(this);
        findViewById(R.id.bt_notification).setOnClickListener(this);
        findViewById(R.id.bt_view_drag_helper_simple).setOnClickListener(this);
        findViewById(R.id.bt_view_drag_helper_simple_1).setOnClickListener(this);
        findViewById(R.id.bt_sensor).setOnClickListener(this);
        findViewById(R.id.bt_glide).setOnClickListener(this);
        findViewById(R.id.bt_keyboard).setOnClickListener(this);
    }

    @Override
    protected boolean setFullScreen() {
        return false;
    }

    @Override
    protected String setTag() {
        return getClass().getSimpleName();
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.activity_function_menu;
    }

    @Override
    public void onClick(View v) {
        Intent intent = null;
        switch (v.getId()) {
            case R.id.bt_gesto:
                intent = new Intent(getActivity(), GestoActivity.class);
                break;
            case R.id.bt_simple_finger_gesture:
                intent = new Intent(getActivity(), SimpleFingerGesturesActivity.class);
                break;
            case R.id.bt_hashmap:
                intent = new Intent(getActivity(), HashMapActivity.class);
                break;
            case R.id.bt_drag_drop_sample:
                intent = new Intent(getActivity(), DragDropSampleActivity.class);
                break;
            case R.id.bt_toggle_fullscreen:
                intent = new Intent(getActivity(), FullScreenActivity.class);
                break;
            case R.id.bt_view_drag_helper:
                intent = new Intent(getActivity(), ViewDragHelperActivity.class);
                break;
            case R.id.bt_recolor:
                intent = new Intent(getActivity(), RecolorActivity.class);
                break;
            case R.id.bt_activity_service_comunicate:
                intent = new Intent(getActivity(), ActivityServiceComunicateActivity.class);
                break;
            case R.id.bt_location:
                intent = new Intent(getActivity(), LocationActivity.class);
                break;
            case R.id.bt_notification:
                intent = new Intent(getActivity(), MenuNotificationActivity.class);
                break;
            case R.id.bt_view_drag_helper_simple:
                intent = new Intent(getActivity(), ViewDragHelperSimpleActivity.class);
                break;
            case R.id.bt_view_drag_helper_simple_1:
                intent = new Intent(getActivity(), ViewDragHelperSimpleActivity1.class);
                break;
            case R.id.bt_sensor:
                intent = new Intent(getActivity(), SensorActivity.class);
                break;
            case R.id.bt_glide:
                intent = new Intent(getActivity(), GlideActivity.class);
                break;
            case R.id.bt_keyboard:
                intent = new Intent(getActivity(), KeyboardActivity.class);
                break;
        }
        if (intent != null) {
            startActivity(intent);
            LActivityUtil.tranIn(getActivity());
        }
    }
}

File: app/src/main/java/vn/loitp/app/activity/function/MenuFunctionActivity.kt
package vn.loitp.app.activity.function

import android.os.Bundle
import android.view.View
import androidx.core.view.isVisible
import com.loitpcore.annotation.IsAutoAnimation
import com.loitpcore.annotation.IsFullScreen
import com.loitpcore.annotation.LogTag
import com.loitpcore.core.base.BaseFontActivity
import com.loitpcore.core.utilities.LUIUtil
import kotlinx.android.synthetic.main.activity_menu_function.*
import vn.loitp.app.R
import vn.loitp.app.activity.function.activityAndService.ActivityServiceCommunicateActivity
import vn.loitp.app.activity.function.dragDropSample.DragDropSampleActivity
import vn.loitp.app.activity.function.fullScreen.FullScreenActivity
import vn.loitp.app.activity.function.glide.GlideActivity
import vn.loitp.app.activity.function.hashmap.HashMapActivity
import vn.loitp.app.activity.function.idleTime.IdleTimeActivity
import vn.loitp.app.activity.function.keyboard.KeyboardActivity
import vn.loitp.app.activity.function.keyboardHeightProvider.KeyboardHeightProviderActivity
import vn.loitp.app.activity.function.location.LocationActivity
import vn.loitp.app.activity.function.notification.MenuNotificationActivity
import vn.loitp.app.activity.function.pump.PumpActivity
import vn.loitp.app.activity.function.recolor.RecolorActivity
import vn.loitp.app.activity.function.sensor.SensorActivity
import vn.loitp.app.activity.function.simpleFingerGestures.SimpleFingerGesturesActivity
import vn.loitp.app.activity.function.theme.ThemeActivity
import vn.loitp.app.activity.function.viewDragHelper.ViewDragHelperActivity
import vn.loitp.app.activity.function.viewDragHelperSimple.ViewDragHelperSimpleActivity
import vn.loitp.app.activity.function.viewDragHelperSimple.ViewDragHelperSimpleActivity1
import vn.loitp.app.activity.function.wallpo.WallpoActivity

@LogTag("MenuFunctionActivity")
@IsFullScreen(false)
@IsAutoAnimation(true)
class MenuFunctionActivity : BaseFontActivity(), View.OnClickListener {

    override fun setLayoutResourceId(): Int {
        return R.layout.activity_menu_function
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setupViews()
    }

    private fun setupViews() {
        lActionBar.apply {
            LUIUtil.setSafeOnClickListenerElastic(
                view = this.ivIconLeft,
                runnable = {
                    onBaseBackPressed()
                }
            )
            this.ivIconRight?.setImageResource(R.color.transparent)
            this.viewShadow?.isVisible = false
            this.tvTitle?.text = MenuFunctionActivity::class.java.simpleName
        }
        btSimpleFingerGesture.setOnClickListener(this)
        btHashMap.setOnClickListener(this)
        btIdleTime.setOnClickListener(this)
        btDragDropSample.setOnClickListener(this)
        btToggleFullScreen.setOnClickListener(this)
        btViewDragHelper.setOnClickListener(this)
        btRecolor.setOnClickListener(this)
        btActivityServiceComunicate.setOnClickListener(this)
        btLocation.setOnClickListener(this)
        btNotification.setOnClickListener(this)
        btPump.setOnClickListener(this)
        btViewDragHelperSimple.setOnClickListener(this)
        btViewDragHelperSimple1.setOnClickListener(this)
        btSensor.setOnClickListener(this)
        btGlide.setOnClickListener(this)
        btKeyboard.setOnClickListener(this)
        btKeyboardHeightProvider.setOnClickListener(this)
        btTheme.setOnClickListener(this)
        btWallpoActivity.setOnClickListener(this)
    }

    override fun onClick(v: View) {
        when (v) {
            btSimpleFingerGesture -> launchActivity(SimpleFingerGesturesActivity::class.java)
            btHashMap -> launchActivity(HashMapActivity::class.java)
            btIdleTime -> launchActivity(IdleTimeActivity::class.java)
            btDragDropSample -> launchActivity(DragDropSampleActivity::class.java)
            btToggleFullScreen -> launchActivity(FullScreenActivity::class.java)
            btViewDragHelper -> launchActivity(ViewDragHelperActivity::class.java)
            btRecolor -> launchActivity(RecolorActivity::class.java)
            btActivityServiceComunicate -> launchActivity(ActivityServiceCommunicateActivity::class.java)
            btLocation -> launchActivity(LocationActivity::class.java)
            btNotification -> launchActivity(MenuNotificationActivity::class.java)
            btPump -> launchActivity(PumpActivity::class.java)
            btViewDragHelperSimple -> launchActivity(ViewDragHelperSimpleActivity::class.java)
            btViewDragHelperSimple1 -> launchActivity(ViewDragHelperSimpleActivity1::class.java)
            btSensor -> launchActivity(SensorActivity::class.java)
            btGlide -> launchActivity(GlideActivity::class.java)
            btKeyboard -> launchActivity(KeyboardActivity::class.java)
            btKeyboardHeightProvider -> launchActivity(KeyboardHeightProviderActivity::class.java)
            btTheme -> launchActivity(ThemeActivity::class.java)
            btWallpoActivity -> launchActivity(WallpoActivity::class.java)
        }
    }
}
